providers/digitalocean/create.go

```
@@ -105,7 +105,6 @@ func CreateDigitaloceanCluster(definition *pkgtypes.ClusterDefinition) error {
 	secretsFilesToRestore, err := os.ReadDir(ctrl.ProviderConfig.SSLBackupDir + "/secrets")
 	if err != nil {
 		log.Info().Msg(err.Error())
-		return fmt.Errorf("error reading secrets directory for TLS: %w", err)
```

Are you sure this should swallow the error?

---

yess i tested,it works :)

---

Can you explain the reason why the error is ignored, please? I'm always very nervous when I see an ignored error.

---

the secrets folder is created when we do backup cmd and not when we provision it,i think we can fully remove the secrets check from create.go of all cloud

---

@mrsimonemms I think it's fine to not return the error. Not every setup uses SSL with certs managed by the cluster (for example, if you use cloudflare) so this could fail on those situations but it isn't really a failure. Good catch though!

---

Can we patch the line right above @jokestax? Something like this for the next person like Simon or me that might think returning the error makes sense:

```diff
-		log.Info().Msg(err.Error())
+		log.Info().Msgf("ignoring error: directory %q couldn't be stat", ctrl.ProviderConfig.SSLBackupDir + "/secrets")
```

I'll try to come back to this and report the error as `not found` vs having other stat issues.